src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -766,7 +766,20 @@ public void setKeepSpawnInMemory(boolean keepLoaded) {
             // find a spawn if needed
             Location spawn = generator.getFixedSpawnLocation(this, random);
             // we're already going to anchor if told to, so don't request another anchor
-            setSpawnLocation(spawn.getBlockX(), spawn.getBlockY(), spawn.getBlockZ(), false);
+            if (spawn == null) {
+                // determine a location randomly
+                int spawnX = random.nextInt(256) - 128, spawnZ = random.nextInt(256) - 128;
+                getChunkAt(spawnX >> 4, spawnZ >> 4).load(true);  // I'm not sure there's a sane way around this
+
+                for (int tries = 0; tries < 1000 && !generator.canSpawn(this, spawnX, spawnZ); ++tries) {
```

Comment:
Because that method isn't static.

I didn't change the value because that's outside of the scope of the PR. I could expand the scope to change the loop to a more appropriate value, and add a static helper method for default random spawn finding code.